cmd/local.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"bytes"
+	"github.com/falcosecurity/driverkit/pkg/driverbuilder"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"golang.org/x/sys/unix"
+	"log/slog"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// NewLocalCmd creates the `driverkit local` command.
+func NewLocalCmd(rootCommand *RootCmd, rootOpts *RootOptions, rootFlags *pflag.FlagSet) *cobra.Command {
+	localCmd := &cobra.Command{
+		Use:               "local",
+		Short:             "Build Falco kernel modules and eBPF probes in local env with local kernel sources and gcc/clang.",
+		PersistentPreRunE: persistentPreRunFunc(rootCommand, rootOpts),
+		Run: func(c *cobra.Command, args []string) {
+			slog.With("processor", c.Name()).Info("driver building, it will take a few seconds")
+			if !configOptions.DryRun {
+				b := rootOpts.ToBuild()
+				if !b.HasOutputs() {
+					return
+				}
+				if err := driverbuilder.NewLocalBuildProcessor(viper.GetInt("timeout")).Start(b); err != nil {
+					slog.With("err", err.Error()).Error("exiting")
+					os.Exit(1)
+				}
+			}
+		},
+	}
+	// Add root flags, but not the ones unneeded
+	unusedFlagsSet := map[string]struct{}{
+		"architecture":        {},
+		"kernelrelease":       {},
+		"kernelversion":       {},
+		"target":              {},
+		"kernelurls":          {},
+		"builderrepo":         {},
+		"builderimage":        {},
+		"gccversion":          {},
+		"kernelconfigdata":    {},
+		"proxy":               {},
+		"registry-name":       {},
+		"registry-password":   {},
+		"registry-plain-http": {},
+		"registry-user":       {},
+	}
+	flagSet := pflag.NewFlagSet("local", pflag.ExitOnError)
+	rootFlags.VisitAll(func(flag *pflag.Flag) {
+		if _, ok := unusedFlagsSet[flag.Name]; !ok {
+			flagSet.AddFlag(flag)
+		}
+	})
+	localCmd.PersistentFlags().AddFlagSet(flagSet)
+	return localCmd
+}
+
+// Partially overrides rootCmd.persistentPreRunFunc setting some defaults before config init/validation stage.
+func persistentPreRunFunc(rootCommand *RootCmd, rootOpts *RootOptions) func(c *cobra.Command, args []string) error {
```

Comment:
This gets called in place of `rootCmd.PersistentPreRunE` since we shadowed it.
It does just set a bunch of default values (that are not used anyway by the `local` build processor, but are logged) and calls the `rootCmd.PersistentPreRunE` (ie: persistentValidateFunc)